### PR TITLE
MAGECLOUD-3470: Add validator for SearchEngine config integrity

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description

Added validation for SEARCH_CONFIGURATION variable for containing at least `engine` option if merge not required.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3470

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-57  last 3 steps

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Config/Validator/Deploy/SearchConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        $searchConfig = $this->stageConfig->get(DeployInterface::VAR_SEARCH_CONFIGURATION);
+        if ($this->configMerger->isEmpty($searchConfig) || $this->configMerger->isMergeRequired($searchConfig)) {
+            return $this->resultFactory->success();
+        }
+
+        if (!isset($searchConfig['engine'])) {
+            return $this->resultFactory->error(
+                sprintf('Variable %s is not configured properly', DeployInterface::VAR_SEARCH_CONFIGURATION),
+                'At least engine option must be configured'
```
Comment (shiftedreality) [inline src/Config/Validator/Deploy/SearchConfiguration.php:66]: add double quotes to highlight `engine` word

File: src/Test/Unit/Config/Validator/Deploy/SearchConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Test\Unit\Config\Validator\Deploy;
```
Comment (shiftedreality) [inline src/Test/Unit/Config/Validator/Deploy/SearchConfigurationTest.php:8]: Add `strict_types` declaration to all new classes

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Test/Unit/Config/Validator/Deploy/SearchConfigurationTest.php (hunk 2 above)
Comment (oshmyheliuk) [reply src/Test/Unit/Config/Validator/Deploy/SearchConfigurationTest.php:8]: Done

File: src/Config/Validator/Deploy/SearchConfiguration.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Config/Validator/Deploy/SearchConfiguration.php:66]: Done

**Review by shiftedreality — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA Aproved

**Pull request merged**